web/src/main/java/org/mskcc/cbio/portal/web/ProxySessionServiceController.java

```diff
@@ -75,16 +93,33 @@
         }
         return null;
     }
+
+    public HttpHeaders createHeaders(){
```

Comment:
may be we have condition inside the object initialization, something like
```
private HttpHeaders createHeaders(){
	return new HttpHeaders() {{
		if (sessionServiceUser != null && !sessionServiceUser.equals("") && sessionServicePassword != null && !sessionServicePassword.equals("")) {
			String auth = sessionServiceUser + ":" + sessionServicePassword;
			byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(Charset.forName("US-ASCII")));
			String authHeader = "Basic " + new String(encodedAuth);
			set( "Authorization", authHeader);
		}
		set( "Content-Type", "application/json");
	}};
}
```
and no need to check for null at the methods invoking it

Reply:
@kalletlak 	good point - thanks for reviewing! Updated it